### PR TITLE
get more info from mill test

### PR DESCRIPTION
**Pull request by kasiaMarek**

### CODE REVIEW

File: tests/slow/src/test/scala/tests/mill/MillServerSuite.scala [hunk 1]
```diff
@@ -24,11 +24,22 @@ class MillServerSuite
...
   val buildTool: MillBuildTool = MillBuildTool(() => userConfig)
+  def bspTrace: AbsolutePath =
+    workspace.resolve(".metals").resolve("bsp.trace.json")
 
   override def currentDigest(
       workspace: AbsolutePath
   ): Option[String] = MillDigest.current(workspace)
 
+  override def afterEach(context: AfterEach): Unit = {
+    if (context.test.name == s"generate-and-connect-$supportedBspVersion") {
```
Comment (tgodzik) [inline tests/slow/src/test/scala/tests/mill/MillServerSuite.scala:35]: The trace needs to be create first, so you would also need to implement afterEach. Maybe instead you could just create a method `testWithBsp` that would wrap the test itself and create the bsp trace before and print it in a catch clause?

This way we could reuse the method and not need to match on the test name

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: tests/slow/src/test/scala/tests/mill/MillServerSuite.scala (hunk 1 above)
Comment (kasiaMarek) [reply tests/slow/src/test/scala/tests/mill/MillServerSuite.scala:35]: The problem is that I want to print it even on timeout, which is put on the whole test. I think only doing it `after` the test will work in that situation. I guess I can not use using `BaseFunSuite.test` and instead add the test to buffer by hand...?

Comment (tgodzik) [reply tests/slow/src/test/scala/tests/mill/MillServerSuite.scala:35]: We have a very similar thing in `BaseDapSuite`, you can add a test transform by overriding munitTestTransforms

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**